src/robot/libraries/dialogs_py.py

```diff
@@ -154,10 +154,15 @@ def _get_value(self) -> str:
 
 class SelectionDialog(TkDialog):
 
-    def _create_widget(self, parent, values) -> Listbox:
+    def _create_widget(self, parent, values, default: Union[int,str,None]=None) -> Listbox:
         widget = Listbox(parent)
         for item in values:
             widget.insert(END, item)
+        if default is not None:
+            if isinstance(default,int):
+                widget.select_set(values[default])
+            elif isinstance(default,str):
+                widget.select_set(values.index[default])
```

Comment:
- As far as I can see, there has been no integer conversion earlier, so if there's a usage like `default=1`, the value is a string.
- As discussed earlier, indexing should start from 1, not from 0.
- The above `widget.select_set` usages get different values as argument. In the first usage the value is one of the original values (i.e. a string) but in the latter case its an integer returned by `values.index()`. I haven't looked at does `select_set` actually accept different values like this, but this doesn't look correct to me. It seems tests are missing or haven't been run properly.
- There's no error checking for invalid default values. The code should handle a situation where the index is out-of-bounds or the default value isn't in the original values so that there's a clear error message. These situations also need to be tested.
- There should be a space after a comma like `isinstance(default, int)`.

Reply:
[aef0c56](https://github.com/robotframework/robotframework/pull/5182/commits/aef0c56dc9b11c5562939bf7cce71072a922b8da)